src/ethereum/shanghai/fork_types.py

```diff
@@ -210,7 +209,26 @@ class Header:
     nonce: Bytes8
     base_fee_per_gas: Uint
     withdrawals_root: Root
+    inclusion_list_summary_root: Root     # [New in EIP7547]
+
 
+@dataclass
+class InclusionListSummaryEntry:
+    """
+    A commitment to an address and nonce pair.
+    """
+    address: Address
+    nonce: U64
+
+@dataclass
+class InclusionListSummary:
+    """
+    Full inclusion list summary.
+    """
+    slot: U64
+    proposer_index: U64
+    parent_hash: Hash32
```

Comment:
Do we need to have `nonce` or only `address` list should suffice as inclusion entries?
I don't see any use on EL side for `slot`, `proposer_index` nor `parent_hash`, why are they supplied?

Reply:
nonce is used to avoid having to fetch which entries were satisfied by the parent block. danny prefers to not have cross block dependencies (e.g., need to fetch n-1 block to validate block n). 

the slot and proposer will be checked by the CL but to make the type the same i just pass them in. for parent_hash, we need to make sure the parent hash matches. let me try to add that somewhere

Reply:
https://github.com/michaelneuder/execution-specs/pull/1/commits/596f3cad972cb60eed3e770bb2811e08e3338afa

Reply:
I probably don't understand something, but this seems overly complicated to me.
Currently IL requires:
- That a block is built on specified parent block
- That a block has transactions for address X up to at leas nonce N.

Why can't it be simplified to only be:
- Block has transactions for address X?